src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -359,28 +353,27 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
   // Internal state, for memoization purposes only
   const [walletType, setWalletType] = useState<KeyManagerWithProvider | null>(null)
 
-  const getAdapter = useCallback(
-    async (keyManager: KeyManager, index: number = 0) => {
-      let currentStateAdapters = state.adapters ?? new Map<KeyManager, any[]>()
+  const getAdapter: GetAdapter = useCallback(
+    async (keyManager, index: number = 0) => {
+      let currentStateAdapters = state.adapters ?? ({} as AdaptersByKeyManager)
 
       // Check if adapter is already in the state
-      let adapterInstance = currentStateAdapters.get(keyManager)?.[index]
-      const currentKeyManagerAdapters = currentStateAdapters.get(keyManager) ?? []
+      let adapterInstance = currentStateAdapters[keyManager]?.[index]
+      const currentKeyManagerAdapters = currentStateAdapters[keyManager] ?? []
 
       if (!adapterInstance) {
         // If not, create a new instance of the adapter
         try {
           const Adapter = await SUPPORTED_WALLETS[keyManager].adapters[index].loadAdapter()
+          const keyManagerOptions = getKeyManagerOptions(keyManager, isDarkMode)
+          // @ts-ignore tsc is drunk as well, not narrowing to the specific adapter and its KeyManager options here
           // eslint is drunk, this isn't a hook
           // eslint-disable-next-line react-hooks/rules-of-hooks
-          adapterInstance = Adapter.useKeyring(
-            state.keyring,
-            getKeyManagerOptions(keyManager, isDarkMode),
-          )
+          adapterInstance = Adapter.useKeyring(state.keyring, keyManagerOptions)
 
           if (adapterInstance) {
             currentKeyManagerAdapters[index] = adapterInstance
```

Comment:
not introduced in this pr, but guh this should at least be pushing to the array (not indexing it like an object)

Reply:
we'd be better off removing the array aspect of this interface and just enforce a since adapter per keymanager